src/StateService/MPT/MPTTrie.Put.cs

```diff
@@ -23,12 +23,8 @@ public bool Put(TKey key, TValue value)
         {
             var path = ToNibbles(key.ToArray());
             var val = value.ToArray();
-            if (path.Length == 0 || path.Length > MPTNode.MaxKeyLength)
+            if (path.Length > MPTNode.MaxKeyLength || val.Length > MPTNode.MaxValueLength)
```

Comment:
Technically, we can allow having empty paths in MPT and MPT will be OK with that. But practically, where do empty paths come from? Even if `System.Storage.Put` allows to use an empty key, all storage items have at least contract ID bytes as the key prefix, so the storage item key is always non-empty. Thus, if MPT should put an item with an empty key, then it's a bug somewhere.

I consider it's safer to keep an old behaviour for the `path` (do not allow 0-lenght `path`).

Reply:
Done.